blockchain/tx_pool.go

```diff
@@ -1531,6 +1550,26 @@ func (pool *TxPool) demoteUnexecutables() {
 			}
 		}
 
+		// TODO : Need to KIP-71 hardfork.
+		// Enqueue transaction if gasPrice of transaction is lower than gasPrice of txPool.
+		// All transactions with a nonce greater than enqueued transaction also stored queue.
+		if list.Len() > 0 {
+			needRemoved := false
```

Comment:
It seems that you use `needRemoved` variable to remove out the txs which has bigger nonce than the tx which has invalid basefee. I think this is similar concept to the `strict`. 
https://github.com/klaytn/klaytn/blob/ed593ea89f26856a77cf78379a91d7abefd933ed/blockchain/tx_pool.go#L942
Can this concept be used instead of implementing remove-out logic at here?

Reply:
I think so. If `tx.GasPrice( )` < `pool.gasPrice`, you don't need to remove `txs` after the `tx's nonce`
Therefore the log, `Demotinc the transaction due to the previous transaction ...` will not appear.
@yoomee1313 @sirano11 How do you think about following code?
```go
if list.Len() > 0 {
	for _, tx := range list.Flatten() {
		hash := tx.Hash()
		if tx.GasPrice().Cmp(pool.gasPrice) < 0 {
		        logger.Trace("Demoting the tx that is lower than the baseFee and those greater than the nonce of the tx.", "txhash", hash)
			list.Remove(tx) // delete all transactions satisfying the nonce value > tx.Nonce( )
			pool.enqueueTx(hash, tx)
			break
		}
	}
}
```

Reply:
I think if you want to fix the code like that, you need to enqueue the filtered-out txs(txs which nonce is bigger than a removed tx), too. It seems that the filtered-out txs are returned by the list.Remove(tx).

Reply:
Thanks, I edited the proposal code.

Reply:
It looks nice, could you add the checking logic `if invalids == nil` when `removed` is true? The Remove returns `true,nil` when strict mode is false. I think strict mode is always true at here because the txList is the element of `pending queue`, right?

Reply:
You are right. 
But I think `for _, invalidTx := range invalids` works like `if invalids == nil` when `invalids == nil`
Actually I don't prefer the implicit code above like remove( ) that is complicated. I think it is better to split remove( ) and removeGreaterThanNonce( ).  However this function is used in many place, so I'd like to postpone this refactoring issue later.

Reply:
@kjeom Oh, you are right. Thanks!

Reply:
Thanks!, I'll resolve it and I'm looking forward to discuss it again later